src/adapters/http/download/router.ts

```diff
@@ -1,9 +1,23 @@
 import path from 'path';
 import express from 'express';
+import { constant, pipe } from 'fp-ts/lib/function';
+import * as TE from 'fp-ts/TaskEither';
+import { SystemEnv } from '../../../useCases/SystemEnv';
+import { makeDownloadRecord } from '../../../domain/DownloadRecord';
+import { persistRecord } from '../../../useCases/PersistRecord';
 
-export const makeDownloadDocumentRouter = (): express.Router => {
+const persistDownloadRecord =
```

Comment:
If `persistRecord` fails, should we call `next` as well?

Reply:
Yes, the next function tells to execute the next middleware, so we should call the `next()` also when something is wrong.